[Bug]: Hard server crash when starting an infusion altar with a gadomancy infusion claw

### Salis Arcana Version

1.1.2-GTNH

### Where is the bug?

Salis Arcana

### Describe the bug

I'm not entirely sure what the cause is since I don't want to restart my server dozens of times (due to other bugs I have to usually restart my server 4-6 times to get it to work), but my server crashed the second I started an autocrafted infusion. Given the stacktrace I'm guessing it has to do with the infusion claw.

### To Reproduce

My setup isn't anything fancy. Given that the stacktrace is erroring because of a null player I'm pretty sure this can be reproduced with any infusion claw, but if it's not reproducible let me know.

### Expected Behavior

No crash

### Attachments

```
[22:22:48] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.util.ReportedException: Exception while updating neighbours
	at Launch//net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:631) ~[MinecraftServer.class:?]
	at Launch//net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:334) ~[lt.class:?]
	at Launch//net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:547) ~[MinecraftServer.class:?]
	at Launch//net.minecraft.server.MinecraftServer.run(MinecraftServer.java:427) [MinecraftServer.class:?]
	at Launch//net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [li.class:?]
Caused by: java.lang.NullPointerException: Cannot read field "field_71075_bZ" because "player" is null
	at Launch//thaumcraft.common.items.wands.ItemWandCasting.handler$bcc000$salisarcana$consumeAllVisOverride(ItemWandCasting.java:776) ~[ItemWandCasting.class:?]
	at Launch//thaumcraft.common.items.wands.ItemWandCasting.consumeAllVis(ItemWandCasting.java) ~[ItemWandCasting.class:?]
	at Launch//makeo.gadomancy.common.blocks.tiles.TileInfusionClaw.hasSufficientVis(TileInfusionClaw.java:379) ~[TileInfusionClaw.class:?]
	at Launch//makeo.gadomancy.common.blocks.tiles.TileInfusionClaw.startClickBlock(TileInfusionClaw.java:255) ~[TileInfusionClaw.class:?]
	at Launch//makeo.gadomancy.common.blocks.tiles.TileInfusionClaw.updateRedstone(TileInfusionClaw.java:176) ~[TileInfusionClaw.class:?]
	at Launch//makeo.gadomancy.common.blocks.BlockInfusionClaw.func_149695_a(BlockInfusionClaw.java:44) ~[BlockInfusionClaw.class:?]
	at Launch//net.minecraft.world.World.func_147460_e(World.java:697) ~[ahb.class:?]
	at Launch//net.minecraft.world.World.func_147459_d(World.java:652) ~[ahb.class:?]
	at Launch//net.minecraft.world.World.func_147444_c(World.java:613) ~[ahb.class:?]
	at Launch//net.minecraft.world.World.markAndNotifyBlock(World.java:488) ~[ahb.class:?]
	at Launch//net.minecraft.world.World.func_147465_d(World.java:466) ~[ahb.class:?]
	at Launch//net.minecraft.block.BlockRedstoneTorch.func_149674_a(SourceFile:136) ~[anc.class:?]
	at Launch//net.minecraft.world.WorldServer.func_72955_a(WorldServer.java:550) ~[mt.class:?]
	at Launch//net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:181) ~[mt.class:?]
	at Launch//net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:625) ~[MinecraftServer.class:?]
	... 4 more
```

### Additional Context

_No response_

--- a/src/main/java/dev/rndmorris/salisarcana/mixins/late/items/MixinItemWandCasting.java
+++ b/src/main/java/dev/rndmorris/salisarcana/mixins/late/items/MixinItemWandCasting.java
@@ -4,9 +4,10 @@ import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 
 import org.spongepowered.asm.mixin.Mixin;
-import org.spongepowered.asm.mixin.injection.At;
-import org.spongepowered.asm.mixin.injection.Inject;
-import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.Unique;
+
+import com.llamalad7.mixinextras.injector.wrapmethod.WrapMethod;
+import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
 
 import thaumcraft.api.aspects.Aspect;
 import thaumcraft.api.aspects.AspectList;
@@ -15,30 +16,35 @@ import thaumcraft.common.items.wands.ItemWandCasting;
 @Mixin(value = ItemWandCasting.class, remap = false)
 public class MixinItemWandCasting {
 
-    @Inject(method = "consumeVis", at = @At(value = "HEAD"), cancellable = true)
-    private void consumeVisOverride(ItemStack is, EntityPlayer player, Aspect aspect, int amount, boolean crafting,
-        CallbackInfoReturnable<Boolean> cir) {
-        if (player.capabilities.isCreativeMode) {
-            cir.setReturnValue(true);
-            cir.cancel();
-        }
+    @Unique
+    private static boolean sa$isCreative(EntityPlayer player) {
+        return player != null && player.capabilities != null && player.capabilities.isCreativeMode;
     }
 
-    @Inject(method = "consumeAllVis", at = @At(value = "HEAD"), cancellable = true)
-    private void consumeAllVisOverride(ItemStack is, EntityPlayer player, AspectList aspects, boolean doit,
-        boolean crafting, CallbackInfoReturnable<Boolean> cir) {
-        if (player.capabilities.isCreativeMode) {
-            cir.setReturnValue(true);
-            cir.cancel();
+    @WrapMethod(method = "consumeVis")
+    private boolean wrapConsumeVis(ItemStack is, EntityPlayer player, Aspect aspect, int amount, boolean crafting,
+        Operation<Boolean> original) {
+        if (sa$isCreative(player)) {
+            return true;
         }
+        return original.call(is, player, aspect, amount, crafting);
     }
 
-    @Inject(method = "consumeAllVisCrafting", at = @At(value = "HEAD"), cancellable = true)
-    private void consumeVisOverride(ItemStack is, EntityPlayer player, AspectList aspects, boolean doit,
-        CallbackInfoReturnable<Boolean> cir) {
-        if (player.capabilities.isCreativeMode) {
-            cir.setReturnValue(true);
-            cir.cancel();
+    @WrapMethod(method = "consumeAllVis")
+    private boolean wrapConsumeAllVis(ItemStack is, EntityPlayer player, AspectList aspects, boolean doit,
+        boolean crafting, Operation<Boolean> original) {
+        if (sa$isCreative(player)) {
+            return true;
         }
+        return original.call(is, player, aspects, doit, crafting);
+    }
+
+    @WrapMethod(method = "consumeAllVisCrafting")
+    private boolean wrapConsumeAllVisCrafting(ItemStack is, EntityPlayer player, AspectList aspects, boolean doit,
+        Operation<Boolean> original) {
+        if (sa$isCreative(player)) {
+            return true;
+        }
+        return original.call(is, player, aspects, doit);
     }
 }

